src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -246,11 +253,33 @@ async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonS
                 "`[python-setup].resolve_all_constraints` is enabled, so "
                 "`[python-setup].requirement_constraints` must also be set."
             )
+        elif request.resolve_and_lockfile:
+            resolve, lockfile = request.resolve_and_lockfile
+            resolved_dists = await Get(
+                ResolvedDistributions,
+                PexRequest(
+                    description=f"Installing {lockfile} for the resolve `{resolve}`",
+                    output_filename=f"{path_safe(resolve)}_lockfile.pex",
+                    internal_only=request.internal_only,
+                    requirements=Lockfile(
+                        file_path=lockfile,
+                        file_path_description_of_origin=(
+                            f"the resolve `{resolve}` (from "
+                            "`[python-setup].experimental_resolves_to_lockfiles`)"
+                        ),
+                        # TODO(#12314): Hook up lockfile staleness check.
+                        lockfile_hex_digest=None,
+                    ),
+                    interpreter_constraints=interpreter_constraints,
+                    platforms=request.platforms,
+                    additional_args=request.additional_args,
+                ),
+            )
         elif python_setup.lockfile:
             resolved_dists = await Get(
                 ResolvedDistributions,
                 PexRequest(
-                    description=f"Resolving {python_setup.lockfile}",
+                    description=f"Installing {python_setup.lockfile}",
```

Comment:
"Installing" sounds like a sideeffect to me, but I guess that it is overloaded between "building a wheel" and "installing something somewhere".

Reply:
@benjyw thoughts?